File: pokemon-chart-api/src/DbRepository/Access/PokemonTypeAccess.cs
using Microsoft.EntityFrameworkCore;
using System.Threading.Tasks;
using System.Collections.Generic;


namespace DbRepository
{
    public class PokemonTypeDbAccess
    {
        PokemonDbContext _c;

        public PokemonTypeDbAccess(PokemonDbContext c)
        {
            this._c = c;

        }

        public async Task<PokemonType> getPokemonType(int id)
        {
            var r = await
                    _c
                        .PokemonTypes
                        .SingleOrDefaultAsync(q => q.Id == id);
            return r;
        }

        public async Task<List<PokemonType>> getAllPokemonTypes()
        {
            var r = await
                    _c
                        .PokemonTypes
                        .ToListAsync();
            return r;
        }

    }
}

File: pokemon-chart-api/src/DbService/Access/PokemonTypeAccess.fs
namespace DbService

    module PokemonTypeAccess =
        open DbContext
        open FSharp.Control.Tasks.V2.ContextInsensitive
        open DbRepository
        open PokemonTypeMappers

        let private _getPokemonTypeById (ctx: PokemonDbContext) id =
            task{
                let p = PokemonTypeDbAccess(ctx)
                let! qr = id |> p.getPokemonType
                let r = qr |> mapDbRepoToDbService
                return r
            }
           
        let private _getAllPokemonTypes(ctx: PokemonDbContext)=
            task{
                let p = PokemonTypeDbAccess(ctx)
                let! qr = p.getAllPokemonTypes()
                let r = qr |> Seq.map mapDbRepoToDbService
                return r
            }
        let getPokemonTypeById = _getPokemonTypeById
        let getAllPokemonTypes = _getAllPokemonTypes

